### PR TITLE
[MNT] update pre commit hook versions and corresponding changes

### LINKED ISSUE #4676: [MNT] update pre-commit hook versions and make appropriate changes in the affected files
Most hook versions in [.pre-commit-config.yaml](https://github.com/sktime/sktime/blob/main/.pre-commit-config.yaml) is quite outdated. These may be updated to the latest verions using `pre-commit autoupdate`.

However, it leads to formatting changes (mostly by `black`) to ~140 files, as per `pre-commit run --all-files`. This will lead to change in Git blame for all of these. While that is useful to keep track of changes and ask for help/guidance in future by newer contributors, this may be worth considering in future.

Ref. https://github.com/sktime/sktime/pull/4673#issuecomment-1575415066

### PR DESCRIPTION
**Pull request by yarnabrina**
Closes #4676.

What this PR does:

1. Update hook versions.
2. Apply code formatting changes identified by `black`. (affected ~140 files)
3. Apply docstring length/style changes identified by `flake8`+`pydocstyle`. (affected ~180 files)

Notes for reviewers:

* For the 3rd point above, used `docformatter` locally to automate few docstring changes, and these are part of d736b6624d5e6254da4a7888d491eb157dcb0ccf.
* In few places where it was unable to solve the issues (mostly `D205` and `D400`), used my own judgement to modify docstrings slightly to avoid lint failures. These possibly biased changes are part of ae526adfb2eef63970803bb51d5babe0245438ec, and should be reviewed carefully.

### CODE REVIEW

File: sktime/_contrib/distance_based/_proximity_forest.py [hunk 1]
```diff
@@ -928,7 +924,7 @@ def split_X_per_class(X, y):
         :param X: Array-like containing instances
         :param y: Array-like containing class labels
         :return: Returns a dictionary {Label: [sub_X]} in which sub_X contains all
-        instances that match that label
+                instances that match that label
```
Comment (fkiraly) [inline sktime/_contrib/distance_based/_proximity_forest.py:927]: funny change

File: sktime/_contrib/distance_based/_proximity_forest.py [hunk 2]
```diff
@@ -1083,8 +1079,8 @@ def calculate_distance_to_exemplars(self, X):
         """Find distance to exemplars.
 
         :param X: the dataset containing a list of instances
-        :return: 2d numpy array of distances from each instance to each
-        exemplar (instance by exemplar)
+        :return: 2d numpy array of distances from each instance to each         exemplar
```
Comment (fkiraly) [inline sktime/_contrib/distance_based/_proximity_forest.py:1082]: funny change

File: sktime/_contrib/distance_based/_proximity_forest.py [hunk 3]
```diff
@@ -1539,8 +1535,8 @@ def calculate_distributions(self, predictions_per_tree, size):
         """Find probability estimates for each class for all cases in X.
 
         :param predictions_per_tree: Array-like of shape
-        [n_instances,[n_tree_estimators,labels]] which contains an array of labels
-        predicted by each tree for each instance.
+                [n_instances,[n_tree_estimators,labels]] which contains an array of
```
Comment (fkiraly) [inline sktime/_contrib/distance_based/_proximity_forest.py:1538]: funny change

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks!

* the changes touch a lot of files, although it is mainly linting
* I noticed a few funny changes, not sure what is at cause of them, but it looks in a few instances the formatting gets worse, not better like in the rest - what is happening there?
* also, quick question: is this introducing anything that would break for the user? I guess not, so it can go with 0.19.2, but just checking your understanding.

**Comment by yarnabrina** [comment]
> I noticed a few funny changes, not sure what is at cause of them, but it looks in a few instances the formatting gets worse, not better like in the rest - what is happening there?

The three changes you pointed out are all automated changes, not by myself. I am not very familiar with `sphinx` style docstrings, but I believe that the standard `docformatter` is following is if the description length exceeds configured line length, which is the same as that of `black`, it splits into next line and adds an indentation. The GitHub diff looks weird, but seeing the actual file contents in VS Code makes it clearer.

> is this introducing anything that would break for the user?

I've made no code changes myself. It should be good to go. May be not in this PR, but in a different PR we can consider adding a workflow rule of auto-updating hooks in a schedule to avoid such big PR's in future.

---

I merged `main` and ran the hooks once again. Few more files got changed this time, but mostly whitespace related changes in docstrings. Also replaced curly/smart quotes with straight quotes using `texthooks`. Please take a look.

**Review by fkiraly — APPROVED** [review]
Happy with this.

**Comment by fkiraly** [comment]
(the docstring fixes look fine)

**Pull request merged**